### PR TITLE
Break out common API testing functionality from pkg/master

### PR DESCRIPTION
**Pull request by luxas**
**What type of PR is this?**
> Uncomment only one, leave it on its own line:
>
> /kind api-change
> /kind bug
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

/kind cleanup

**What this PR does / why we need it**:

In order to be able to test any API group if it has the JSON tags set correctly, this piece of code should be in a easily-vendorable package. This code will be referenced from `k8s.io/component-base` testing code as well later on.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```
/assign @sttts @liggitt 
/priority backlog
/milestone v1.14

### CODE REVIEW

File: staging/src/k8s.io/apimachinery/pkg/api/apitesting/naming/naming.go [hunk 1]
```diff
@@ -0,0 +1,142 @@
+/*
```
Comment (sttts) [inline staging/src/k8s.io/apimachinery/pkg/api/apitesting/naming/naming.go]: wouldn't pkg/apitesting or even pkg/api/testing fit better?

Comment (luxas) [reply staging/src/k8s.io/apimachinery/pkg/api/apitesting/naming/naming.go]: This directory already exists. Don't ask me why it's named this way, but I think this place would make the most sense vs creating a new dir.

**Comment by sttts** [comment]
Mostly code move, adapted a bit for cleaner test output.

/lgtm
/approve

**Pull request merged**